server/pages/index.js

```diff
@@ -375,20 +378,44 @@ module.exports = function() {
 		app.get( '/themes/:theme_slug', function( req, res ) {
 			const context = getDefaultContext( req );
 
-			if ( config.isEnabled( 'server-side-rendering' ) ) {
+			const renderThemeSheet = function( theme ) {
 				const store = createReduxStore();
+				store.dispatch( {
+					type: ActionTypes.RECEIVE_THEME_DETAILS,
+					themeId: theme.id,
+					themeName: theme.name,
+					themeAuthor: theme.author,
+					themeScreenshot: theme.screenshot,
+				} );
```

Comment:
Actually, `renderThemeSheet` looks like something we could keep as middleware in `my-sites/themes/controller`: it basically writes to `context` and dispatches stuff via `store`, which should be available via `context`.